linc_convert/modalities/lsm/multi_slice.py

```diff
@@ -0,0 +1,374 @@
+        all_chunks_info["y"].append(int(parsed.group("y")))
+
+    # default output name
+    if not out:
+        out = all_chunks_info["prefix"][0] + all_chunks_info["suffix"][0]
+        out += ".nii.zarr" if nii else ".ome.zarr"
+    nii = nii or out.endswith(".nii.zarr")
+
+    # parse all individual file names
+    nchunkz = max(all_chunks_info["z"])
+    nchunky = max(all_chunks_info["y"])
+    allshapes = [[(0, 0, 0) for _ in range(nchunky)] for _ in range(nchunkz)]
+
+    dtype = None
+    for zchunk in range(nchunkz):
+        for ychunk in range(nchunky):
+            for i in range(len(all_chunks_info["filename"])):
+                if (
+                    all_chunks_info["z"][i] == zchunk + 1
+                    and all_chunks_info["y"][i] == ychunk + 1
+                ):
+                    break
+            filename = all_chunks_info["filename"][i]
+            f = TiffFile(filename)
+            dtype = f.pages[0].dtype
+            allshapes[zchunk][ychunk] = (len(f.pages), *f.pages[0].shape)
+
+    # check that all chunk shapes are compatible
+    for zchunk in range(nchunkz):
+        if len(set(shape[1] for shape in allshapes[zchunk])) != 1:
+            raise ValueError("Incompatible Y shapes")
+    for ychunk in range(nchunky):
+        if len(set(shape[ychunk][0] for shape in allshapes)) != 1:
+            raise ValueError("Incompatible Z shapes")
+    if len(set(shape[2] for subshapes in allshapes for shape in subshapes)) != 1:
+        raise ValueError("Incompatible X shapes")
+
+    # compute full shape
+    fullshape = [0, 0, 0]
+    fullshape[0] = sum(shape[0][0] for shape in allshapes)
+    fullshape[1] = (
+        sum(shape[1] for shape in allshapes[0])
+        - (len(set(all_chunks_info["y"])) - 1) * overlap
+    )
+    fullshape[2] = allshapes[0][0][2]
+
+    # Prepare Zarr group
+    omz = zarr.storage.DirectoryStore(out)
+    omz = zarr.group(store=omz, overwrite=True)
+    print(out)
+    # Prepare chunking options
+    opt = {
+        "chunks": [chunk] * 3,
+        "dimension_separator": r"/",
+        "order": "F",
+        "dtype": np.dtype(dtype).str,
+        "fill_value": None,
+        "compressor": make_compressor(compressor, **compressor_opt),
+    }
+
+    # write first level
+    omz.create_dataset("0", shape=fullshape, **opt)
+    array = omz["0"]
+    print("Write level 0 with shape", fullshape)
+
+    for i, filename in enumerate(all_chunks_info["filename"]):
+        chunkz = all_chunks_info["z"][i] - 1
+        chunky = all_chunks_info["y"][i] - 1
+        f = TiffFile(filename)
+
+        pages = f.pages
+        dat = f.asarray()
+        if len(set(all_chunks_info["y"])) != 1 and overlap!=0:
+            if chunky != 0:
+                dat = dat[:, overlap // 2 :, :]
+            if chunky != max(all_chunks_info["y"])-1:
+                dat = dat[:, :-overlap // 2 - (overlap - 2 * (overlap // 2)), :]
+
+        zstart = sum(shape[0][0] for shape in allshapes[:chunkz])
```

Comment:
I might be mistaken, but I think we need the following change:

```suggestion
        zstart = sum(shape[chunky][0] for shape in allshapes[:chunkz])
```